src/Features/CSharp/Portable/IntroduceVariable/CSharpIntroduceVariableService_IntroduceLocal.cs

```diff
@@ -56,9 +56,12 @@ internal partial class CSharpIntroduceVariableService
                         document, block, expression, newLocalName, declarationStatement, allOccurrences, cancellationToken).ConfigureAwait(false);
 
                 case ArrowExpressionClauseSyntax arrowExpression:
+                    var method = document.SemanticModel.GetDeclaredSymbol(arrowExpression.Parent) as IMethodSymbol;
+                    bool createReturnStatement = !method?.ReturnsVoid ?? true; // this will be null for properties & indexers, both of which return a value
```

Comment:
what if it's the body of a setter?   Or will that give you an IMethodSymbol?  Is htis *only* for properties without accessors that have ```=> expr;```?

Reply:
the setter will give a symbol, getter will too, it's only a top level expression body like this that won't: `int A => 5;`

and there are tests for setters too